src/modules/Hooks.ts

```diff
@@ -16,17 +16,18 @@ export interface HookDefinitions {
 	clickLink: { event: DelegateEvent<MouseEvent> };
 	disabled: undefined;
 	enabled: undefined;
+	fetchPage: { url: string; options: FetchOptions; response?: Response | Promise<Response> };
+	loadPage: { url: string; options: FetchOptions; page?: PageData | Promise<PageData> };
```

Comment:
Are you still planning to refactor `loadPage`, `performPageLoad` et al?

Reply:
Yes, I'd like to do but in another PR. This one is getting a bit too large.